llvm/lib/Analysis/LazyValueInfo.cpp

```diff
@@ -1283,6 +1285,28 @@ std::optional<ValueLatticeElement> LazyValueInfoImpl::getValueFromICmpCondition(
   return ValueLatticeElement::getOverdefined();
 }
 
+ValueLatticeElement LazyValueInfoImpl::getValueFromTrunc(Value *Val,
+                                                         TruncInst *Trunc,
+                                                         bool IsTrueDest) {
+  assert(Trunc->getType()->isIntOrIntVectorTy(1));
+
+  if (Trunc->getOperand(0) != Val)
+    return ValueLatticeElement::getOverdefined();
+
+  Type *Ty = Val->getType();
+
+  if (Trunc->hasNoUnsignedWrap()) {
+    if (IsTrueDest)
+      return ValueLatticeElement::get(
+          Constant::getIntegerValue(Ty, APInt(Ty->getScalarSizeInBits(), 1)));
+    return ValueLatticeElement::get(Constant::getNullValue(Ty));
```

Comment:
You can use `ConstantInt::getBool()` here.

Reply:
I do not think it is not possible to use `ConstantInt::getBool()` as the type of val is not i1.
it is possible to swap the true case to `ConstantInt::get(Ty, 1)`

Reply:
You're right, I got confused with the types here. And yes, using ConstantInt::get is preferable. (getIntegerValue is a specialized API if you need to potentially create an inttoptr expression...)

Reply:
Fixed